src/Krizalys/Onedrive/Client.php

```diff
@@ -618,7 +619,7 @@ public function createFile($name, $parentId = null, $content = '')
 
         // TODO: some versions of cURL cannot PUT memory streams? See here for a
         // workaround: https://bugs.php.net/bug.php?id=43468
-        $file = $this->apiPut($parentId . '/files/' . urlencode($name), $stream);
+        $file = $this->apiPut($parentId . '/files/' . urlencode($name) . '?overwrite=' . ($overwrite ? 'true' : 'ChooseNewName'), $stream);
```

Comment:
Sorry, my mistake not indicate the source, follow the link below. Yes, it
would be interesting to work with the three options, I had opted for two to
be the most used, but there may be cases that need the third option, I do
to change or you do?

https://msdn.microsoft.com/en-us/library/office/dn659726.aspx

Em qui, 26 de mai de 2016 às 14:35, Christophe Vidal <
notifications@github.com> escreveu:

> In src/Krizalys/Onedrive/Client.php
> <https://github.com/krizalys/onedrive-php-sdk/pull/16#discussion_r64787175>
> :
>
> > @@ -618,7 +619,7 @@ public function createFile($name, $parentId = null, $content = '')
> >
> >          // TODO: some versions of cURL cannot PUT memory streams? See here for a
> >          // workaround: https://bugs.php.net/bug.php?id=43468
> > -        $file = $this->apiPut($parentId . '/files/' . urlencode($name), $stream);
> > +        $file = $this->apiPut($parentId . '/files/' . urlencode($name) . '?overwrite=' . ($overwrite ? 'true' : 'ChooseNewName'), $stream);
>
> I did more testing, and I removed my previous comment after having
> discovered the following:
>
>    - passing overwrite=true (or not passing anything at all) will
>    overwrite the file if a file with this name already exists (current
>    behavior)
>    - passing overwrite=false will throw an Exception if a file with this
>    name already exists
>    - passing overwrite=ChooseNewName will create a new file if a file
>    with this name already exists (the file will be named after the original
>    one, followed by a sequence number, such as File 1 if File.
>
> It would be cool to support these 3 policies. Would you like to add
> support for the overwrite=false case and perhaps passing constants such
> as OVERWRITE_ALWAYS, OVERWRITE_NEVER, OVERWRITE_RENAME when calling
> createFile?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/krizalys/onedrive-php-sdk/pull/16/files/29aa7094d8327613fdf94dc066ad7f5ecc3f40b6#r64787175>
>